org.mixedrealitytoolkit.input/Interactors/GazePinch/GazePinchInteractor.cs

```diff
@@ -444,5 +452,64 @@ protected override void OnSelectExited(SelectExitEventArgs args)
         }
 
         #endregion XRBaseInteractor
+
+        /// <summary>
+        /// Updates the pinch state of the GazePinchInteractor.
+        /// If a HandController is present, the pinch state is updated using the HandController's HandNode defaulting to XRNode.RightHand if not set.
+        /// If no HandController is present, the pinch state is updated using the Handedness of the GazePinchInteractor defaulting to InteractorHandedness.Right if not set.
+        /// If the pinch data is not available for the set hand then the other hand is tried.
+        /// </summary>
+        private void UpdatePinchState()
+        {
+            if (logicalSelectState == null)
+            {
+                Debug.LogWarning("GazePinchInteractor is missing logicalSelectState, pinch state won't update.");
+                return;
+            }
+
+            if (XRSubsystemHelpers.HandsAggregator == null)
+            {
+                return;
+            }
+
+            // Determine the XRNode to use for the pinch data
+            bool gotPinchData;
+            XRNode xRNode;
+            if (handController != null) //For pre-XRI3 code.  Note: This if can be removed (leave the code in the 'else' section) when
```

Comment:
We don't need to handle Hand Controller here.  

The else case should work for both rigs.  Have you tried just relying on the stuff in the 'else'?

Reply:
Indeed.  My bad, fixed in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/743/commits/338a5d70ac1bde41db85767a55abd3ee0994d21a , resolving.